app/controllers/batches_controller.rb

```diff
@@ -206,7 +206,9 @@ def fail_items # rubocop:todo Metrics/AbcSize
 
   def sort
     @batch.assign_positions_to_requests!(params['requests_list'].map(&:to_i))
-    @batch.rebroadcast
+
+    # Touch the batch to update its timestamp and trigger re-broadcast
+    @batch.touch # rubocop:disable Rails/SkipsModelValidations
```

Comment:
We were sending the message, but it didn't have an updated timestamp (not to mention was at risk of being generated before the transaction closed). Rather than faffing around pulling in the batch_request timestamps, we just update the batch itself with a touch. Touch in turn triggers rebroadcast at the close of the transaction.